kernel/src/transaction/mod.rs

```diff
@@ -178,6 +181,14 @@ impl Transaction {
         }
     }
 
+    // Set whether this transaction represents a logical data change for the table.
+    // Operations are not a data change when they just reorganize rows within files
+    // but do not add, remove or modify the data in the table otherwise.
+    pub fn with_is_data_change(mut self, is_data_change: bool) -> Self {
```

Comment:
seems reasonable - and this feels like it will be a quick PR :)

nit: I wonder if `with_data_change` would be reasonable? just since with-is feels weird?

Reply:
`with_data_change` seems fine to me.

Reply:
BTW, I think this should be sequenced after remove.  Because right now without a remove API the user has no business setting this (isDataChange should always be true) and we should likely enforce that there are at least some adds and removes provided by the user to allow isDataChange=False